examples/chip-tool-darwin/commands/clusters/ClusterCommandBridge.h

```diff
@@ -0,0 +1,135 @@
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+    }
+
+    ~ClusterCommand() {}
+
+    CHIP_ERROR SendCommand(CHIPDevice * _Nonnull device, chip::EndpointId endpointId) override
+    {
+        chip::TLV::TLVWriter writer;
+        chip::TLV::TLVReader reader;
+        NSDictionary * commandFieldDict = @{@"type" : @"Structure", @"value" : @[]};
+
+        mData = static_cast<uint8_t *>(chip::Platform::MemoryCalloc(sizeof(uint8_t), mDataMaxLen));
+        VerifyOrReturnError(mData != nullptr, CHIP_ERROR_NO_MEMORY);
+
+        writer.Init(mData, mDataMaxLen);
+
+        ReturnErrorOnFailure(mPayload.Encode(writer, chip::TLV::AnonymousTag()));
+        reader.Init(mData, writer.GetLengthWritten());
+        ReturnErrorOnFailure(reader.Next());
+
+        id commandFields = NSObjectFromCHIPTLV(&reader);
+        if (commandFields != nil) {
+            commandFieldDict = (NSDictionary *) commandFields;
+        }
+
+        return ClusterCommand::SendCommand(device, endpointId, mClusterId, mCommandId, commandFieldDict);
+    }
+
+    CHIP_ERROR SendCommand(CHIPDevice * _Nonnull device, chip::EndpointId endpointId, chip::ClusterId clusterId,
+        chip::CommandId commandId, id _Nonnull commandFields)
+    {
+        uint16_t repeatCount = mRepeatCount.ValueOr(1);
+        uint16_t __block responsesNeeded = repeatCount;
+        dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.command", DISPATCH_QUEUE_SERIAL);
+
+        while (repeatCount--) {
+            [device invokeCommandWithEndpointId:[NSNumber numberWithUnsignedShort:endpointId]
+                                      clusterId:[NSNumber numberWithUnsignedInteger:clusterId]
+                                      commandId:[NSNumber numberWithUnsignedInteger:commandId]
+                                  commandFields:commandFields
+                             timedInvokeTimeout:mTimedInteractionTimeoutMs.HasValue()
+                                 ? [NSNumber numberWithUnsignedShort:mTimedInteractionTimeoutMs.Value()]
+                                 : nil
+                                    clientQueue:callbackQueue
+                                     completion:^(
+                                         NSArray<NSDictionary<NSString *, id> *> * _Nullable values, NSError * _Nullable error) {
+                                         CHIP_ERROR err = CHIP_NO_ERROR;
+                                         responsesNeeded--;
+                                         if (error) {
+                                             NSLog(@"Error: %@", error);
+                                         }
+                                         err = [CHIPError errorToCHIPErrorCode:error];
+                                         if (responsesNeeded == 0) {
+                                             SetCommandExitStatus(err);
```

Comment:
What chip-tool does is `SetCommandExitStatus()` with the last error it got if it got any errors, CHIP_NO_ERROR otherwise.  What this code is doing is calling it with the last status we get, which might ignore earlier errors.

Reply:
Ok. Now it will save the last error that occurred in a class variable. I was able to test all error and no error, but I couldn't do  Success, Error, Success.

Reply:
I don't think we have a thing right now that guarantees sometimes error and sometimes success, so testing is a bit painful....

Reply:
Ok.